manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java

```diff
@@ -40,18 +44,51 @@ public void apply(Project project) {
         final ManipulationModel alignmentModel = getCurrentManipulationModel(project.getRootDir());
         final ManipulationModel correspondingModule = alignmentModel.findCorrespondingChild(project.getName());
 
+        final ResolvedDependenciesRepository resolvedDependenciesRepository = new ResolvedDependenciesRepository();
+
+        project.afterEvaluate(p -> {
+            // dependencyManagement is the extension that the Spring Dependency Management Plugin registers
+            final Object obj = p.getExtensions().findByName("dependencyManagement");
+            if (obj != null) {
+                if (isDependencyManagementPluginPomCustomizationEnabled(obj)) {
+                    throw new InvalidUserDataException(
+                            "The ManipulationPlugin cannot be used together with the Spring Dependency Management unless the latter has disabled generatedPomCustomization");
+                }
+            }
+        });
+
         // add actions to manipulate project
         project.afterEvaluate(new ProjectChangeVersionAction(correspondingModule));
-        project.afterEvaluate(new OverrideDependenciesAction(correspondingModule));
+        project.afterEvaluate(new OverrideDependenciesAction(correspondingModule, resolvedDependenciesRepository));
         project.afterEvaluate(new ManifestUpdateAction(correspondingModule));
 
-        configurePublishingTask(project, correspondingModule);
+        configurePublishingTask(project, correspondingModule, resolvedDependenciesRepository);
+    }
+
+    // Ensure that if the Spring Dependency Management plugin is applied,
+    // that it's configured to not generate a "dependencyManagement" section in the generated bom
+    // This is needed because if we don't do it, the "dependencyManagement" section (which is a bom inclusion) will override our dependencies
+    // On the implementation side of things, we need to use reflection because we can get a nasty classloader errors
+    // when trying to cast the object to the known type
```

Comment:
Can you provide some more info on the class loading errors you get?

Reply:
Sure thing. Basically I can't cast to the target class because it gets loaded by a different class loader than this class.
If you have a better suggestion than reflection, I would LOVE to follow it

Reply:
Is it similar to the SLF4J / MDC problem ?

Reply:
It seemed very similar yes